cmd/libsnap-confine-private/infofile.c

```diff
@@ -76,6 +82,31 @@ int sc_infofile_get_key(FILE *stream, const char *key, char **value, sc_error **
         }
         /* Replace the trailing newline character with the NUL byte. */
         line_buf[nread - 1] = '\0';
+
+        /* Handle ini sections (if requested via non-null section name) */
+        if (section != NULL) {
+            if (line_buf[0] == '[') {
+                section_matched = false;
+                char *sec_endptr = memchr(line_buf, ']', nread);
```

Comment:
nread-1 ?, line_buf[nread-1] is always \0 per the code above so we don't need to scan that.